packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js

```diff
@@ -99,6 +104,30 @@ export class MyAccountAddressForm extends FieldForm {
         });
     };
 
+    onZipcodeChange = (e) => {
+        const { value } = e.currentTarget;
+        const { countryId } = this.state;
+
+        try {
+            fetch(`http://api.zippopotam.us/${countryId}/${value}`)
+                .then(
+                /** @namespace Sofacompany/Route/Checkout/Container/fetch/then */
+                    (response) => response.json()
+                )
+                .then(
+                    /** @namespace Sofacompany/Route/Checkout/Container/fetch/then/then */
+                    (data) => {
+                        if (data && Object.entries(data).length > 0) {
+                            this.setState({
+                                city: data.places[0].['place name']
+                            });
+                        }
+                    }
```

Comment:
No data fetches are allowed in the components. If you wish to perform a data fetch on React level - you do that in the container.
But, consider moving this data retrieval into some util function and calling it from there. Who knows where we'll need this in the future :) Let's keep it as reusable as possible.